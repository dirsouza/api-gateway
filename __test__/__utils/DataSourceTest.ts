import { config } from '../../src/config';
import UserAPI from '../../src/dataSources/UserAPI';

export const mocks = {
  get: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  put: jest.fn(),
};

export const API = {
  userAPI: new UserAPI(config),
};
