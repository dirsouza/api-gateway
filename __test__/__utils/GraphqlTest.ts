import { ApolloServer } from 'apollo-server';
import { DataSources } from 'apollo-server-core/dist/requestPipeline';
import { importSchema } from 'graphql-import';
import resolvers from '../../src/resolvers';

import { microService } from '../../src/config';
import * as dataSources from '../../src/dataSources';
import { IDataSources } from '../../src/interfaces';

export const constructTestServer = () => {
  const apis = {
    userApi: new dataSources.UserAPI(microService)
  };

  const server = new ApolloServer({
    typeDefs: importSchema(`${__dirname}/../../src/schemas/index.graphql`),
    resolvers,
    dataSources: (): DataSources<IDataSources> => apis,
    context: () => ({})
  });

  return { server, ...apis };
};
