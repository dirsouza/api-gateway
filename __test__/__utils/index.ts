export * from './DataSourceTest';
export * from './GraphqlTest';
