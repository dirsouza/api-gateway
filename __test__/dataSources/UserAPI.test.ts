import { mocks, API } from '../__utils';

describe('Teste Usuarios DataSource', () => {
  it('return all users', async () => {
    const { userAPI } = API;
    userAPI.getUsers = mocks.get;

    mocks.get.mockReturnValue([{ id: '1', name: 'Teste', email: 'teste' }]);

    const res = await userAPI.getUsers();

    // @ts-ignore
    expect(res).toMatchSnapshot();
    expect(res).toEqual([{ id: '1', name: 'Teste', email: 'teste' }]);
  });

  it('return all users (Exception)', async () => {
    const { userAPI } = API;
    userAPI.getUsers = mocks.get;

    mocks.get.mockReturnValue({ errors: 'Users not found' });

    const res = await userAPI.getUsers();

    expect(res).toEqual({ errors: 'Users not found' });
  });
});
