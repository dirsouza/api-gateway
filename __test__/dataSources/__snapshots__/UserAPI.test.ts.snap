// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste Usuarios DataSource return all users 1`] = `
Array [
  Object {
    "email": "teste",
    "id": "1",
    "name": "Teste",
  },
]
`;
